app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -276,6 +285,107 @@ def parse_namespaces(container_projects)
       parse_base_item(container_projects).except(:namespace)
     end
 
+    def parse_quota(resource_quota)
+      new_result = parse_base_item(resource_quota).except(:namespace)
+      new_result[:container_project] = @data_index.fetch_path(
+        :container_projects,
+        :by_name,
+        resource_quota.metadata["table"][:namespace])
+      new_result[:container_quota_items] = parse_quota_items resource_quota
+      new_result
+    end
+
+    def parse_quota_items(resource_quota)
```

Comment:
@moolitayer I think the whole thing can be simplified to:

    def parse_quota_items(resource_quota)
      new_result_h = Hash.new do |h, k|
        h[k] = {
          :resource       => k.to_s,
          :quota_desired  => nil,
          :quota_enforced => nil,
          :quota_observed => nil
        }
      end
    
      resource_quota.spec.hard.to_h.each do |resource_name, quota|
        new_result_h[resource_name][:quota_desired] = quota
      end
    
      resource_quota.status.hard.to_h.each do |resource_name, quota|
        new_result_h[resource_name][:quota_enforced] = quota
      end
    
      resource_quota.status.used.to_h.each do |resource_name, quota|
        new_result_h[resource_name][:quota_observed] = quota
      end
    
      new_result_h.values
    end

same thing for the limits.